src/parallaximage/ParallaxImage.tsx

```diff
@@ -0,0 +1,260 @@
+// Parallax effect inspired by https://github.com/oblador/react-native-parallax/
+
+import React, { Component } from 'react';
+import {
+    View,
+    Animated,
+    Easing,
+    ActivityIndicator,
+    findNodeHandle,
+    ImageProps,
+    StyleProp,
+    ViewStyle,
+    NativeSyntheticEvent,
+    ImageLoadEventData,
+    ImageErrorEventData
+} from 'react-native';
+import styles from './ParallaxImage.style';
+import type Carousel from 'src/carousel/Carousel';
+
+type ParallaxImageProps = {
+  carouselRef: Carousel<unknown> | null; // passed from <Carousel />
+  itemHeight: number; // passed from <Carousel />
+  itemWidth: number; // passed from <Carousel />
+  scrollPosition: Animated.Value; // passed from <Carousel />
+  sliderHeight: number; // passed from <Carousel />
+  sliderWidth: number; // passed from <Carousel />
+  vertical: boolean; // passed from <Carousel />
+  containerStyle: StyleProp<ViewStyle>;
+  dimensions: {
+    width: number;
+    height: number;
+  };
+  fadeDuration: number;
+  parallaxFactor: number;
+  showSpinner: boolean;
+  spinnerColor: string;
+  AnimatedImageComponent: typeof Animated.Image;
+} & ImageProps;
+
+type ParallaxImageState = {
+  offset: number;
+  width: number;
+  height: number;
+  status: 1 | 2 | 3 | 4;
```

Comment:
I think we can also define the types for these numbers.  It will be easy to understand, when someone is trying to use

Reply:
I tried being as precise as I could and 1|2|3|4 is usable by a function expecting number but it's also more clear

Reply:
I'm ok with keeping the numbers as is. However we should probably copy the following comment there as well for clarity's sake:

```
// 1 -> loading | 2 -> loaded | 3 -> transition finished | 4 -> error
```

Reply:
@bd-arc & @Titozzz we can do one more thing. We can add the meaning of the numbers in the documentation. So that users can understand the meaning of these numbers.

Reply:
I can probably replace with an enum. That'll make things easier

Reply:
@Titozzz Great idea!